### PR TITLE
Add doc examples coverage

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @jyn514

### CODE REVIEW

**Review by jyn514 — COMMENTED** [review]
I'm impressed how little code this is! One small nit, and the build is failing, you might need to run rustfmt or something.

File: src/librustdoc/passes/calculate_doc_coverage.rs [hunk 1]
```diff
@@ -103,40 +121,73 @@ impl CoverageCalculator {
...
     }
 }
 
 impl fold::DocFolder for CoverageCalculator {
     fn fold_item(&mut self, i: clean::Item) -> Option<clean::Item> {
         let has_docs = !i.attrs.doc_strings.is_empty();
+        let mut tests = Tests { found_tests: 0 };
+
+        find_testable_code(
+            &i.attrs.doc_strings.iter().map(|d| d.as_str()).collect::<Vec<_>>().join("\n"),
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/passes/calculate_doc_coverage.rs:183]: We don't clone anything, just create a `Vec<&str>`, so apart from the `Vec` allocation, it doesn't seem that expensive, no?

[the remark this answers is not in the document]
Comment (jyn514) [reply src/librustdoc/passes/calculate_doc_coverage.rs:183]: I'm pretty sure `join()` copies each input. So this creates a Vec with the size of the number of attributes, then copies the full text of each attribute into a larger allocation.

I don't consider this a blocker but I'm curious if you have another way in mind that doesn't require combining all the attributes into one string.

[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/passes/calculate_doc_coverage.rs:183]: Oh you're right! Hum... Let me check if the Markdown parser allows an iterator as an input...

[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/passes/calculate_doc_coverage.rs:183]: Unfortunately, the API seems pretty limited: https://docs.rs/pulldown-cmark/0.7.2/pulldown_cmark/struct.Parser.html

So until it's being extended, I guess we're kinda stuck with this extra allocation...

[the remark this answers is not in the document]
Comment (jyn514) [reply src/librustdoc/passes/calculate_doc_coverage.rs:183]: Got it, I'll open an issue so we don't lose track. Thanks for looking into it :)

**Pull request merged**